wolfcrypt/src/ed25519.c

```diff
@@ -388,18 +396,79 @@ int wc_ed25519ph_sign_msg(const byte* in, word32 inLen, byte* out,
 
 #ifdef HAVE_ED25519_VERIFY
 
+/*
+   sig        is array of bytes containing the signature
+   sigLen     is the length of sig byte array
+   key        Ed25519 public key
+   return     0 on success
+   type       variant to use -- Ed25519, Ed25519ctx, or Ed25519ph
+   context    extra signing data
+   contextLen length of extra signing data
+*/
+int wc_ed25519_verify_msg_init(const byte* sig, word32 sigLen, ed25519_key* key,
+                               byte type, const byte* context, byte contextLen)
+{
+    int    ret;
+
+    /* sanity check on arguments */
+    if (sig == NULL || key == NULL ||
+        (context == NULL && contextLen != 0)) {
+        return BAD_FUNC_ARG;
+    }
+
+    /* check on basics needed to verify signature */
+    if (sigLen != ED25519_SIG_SIZE || (sig[ED25519_SIG_SIZE-1] & 224))
+        return BAD_FUNC_ARG;
+
+    /* find H(R,A,M) and store it as h */
+
+    ret = ed25519_hash_reset(key);
+    if (ret != 0)
+        return ret;
+    if (type == Ed25519ctx || type == Ed25519ph) {
+        ret = ed25519_hash_update(key, ed25519Ctx, ED25519CTX_SIZE);
+        if (ret == 0)
+            ret = ed25519_hash_update(key, &type, sizeof(type));
+        if (ret == 0)
+            ret = ed25519_hash_update(key, &contextLen, sizeof(contextLen));
+        if (ret == 0 && context != NULL)
+            ret = ed25519_hash_update(key, context, contextLen);
+    }
+    if (ret == 0)
+        ret = ed25519_hash_update(key, sig, ED25519_SIG_SIZE/2);
+    if (ret == 0)
+        ret = ed25519_hash_update(key, key->p, ED25519_PUB_KEY_SIZE);
+
+    if (ret < 0)
+        (void)wc_ed25519_verify_msg_final(NULL, 0, NULL, key);
```

Comment:
Why do this?
wc_ed25519_verify_msg_final() immediately returns with BAD_FUNC_ARG.

Reply:
You are absolutely correct.  That call was a vestige of an earlier iteration of this PR, in which `wc_ed25519_verify_msg_final()` was overloaded with the ability to reset the hash state, tolerating null args.  This call has no effect with the current code, and I've removed it.